### PR TITLE
Aryan softplus

### PR DESCRIPTION
**Pull request by Aryan8912**
close : https://github.com/openvinotoolkit/openvino/issues/24109

### CODE REVIEW

**Comment by eshoguli** [comment]
You implemented JIT emitter: good start, thanks! But how did you test it?

I would like to suggest next implementation steps. Explore existing PR from *Example Pull Requests* section here: https://github.com/openvinotoolkit/openvino/issues/24109, please.
1. Update tests to be sure that your implementation is used. Tests should fail: `acl_f32` primitive is used.
2. Update JIT kernel. Tests should fail: expected and reference tensors comparation fails.
3. Finish JIT emitter. Tests should pass.

Let me know if you still have any questions.

**Comment by Aryan8912** [comment]
Hello, sir I do not understand the acl_f32 primitive is used. and also how should Finish JIT emitter

**Comment by eshoguli** [comment]
> Hello, sir I do not understand the acl_f32 primitive is used. and also how should Finish JIT emitter

Hi @Aryan8912, you can use `Sigmoid` implementation as example: https://github.com/openvinotoolkit/openvino/pull/23065. The line where `Sigmoid` is added in `Executer`: https://github.com/openvinotoolkit/openvino/pull/23065/files#diff-ce3f79fe72be04c8660d4f853b11227473660eb109c3fc984c6ff4d2b4d97738R32.

Don't ignore tests, please. Let me know if you still have any questions.

File: src/plugins/intel_cpu/src/emitters/plugin/aarch64/jit_eltwise_emitters.hpp [hunk 1]
```diff
@@ -335,6 +335,45 @@ class jit_select_emitter : public jit_emitter {
     void emit_isa(const std::vector<size_t> &in_vec_idxs, const std::vector<size_t> &out_vec_idxs) const;
 };
 
+class jit_soft_sign_emitter : public jit_emitter {
```
Comment (eshoguli) [inline src/plugins/intel_cpu/src/emitters/plugin/aarch64/jit_eltwise_emitters.hpp:338]: I expect `SoftPlus` declaration here. `SoftSign` should be here: https://github.com/openvinotoolkit/openvino/issues/24114.

File: src/plugins/intel_cpu/src/emitters/plugin/aarch64/jit_eltwise_emitters.cpp [hunk 2]
```diff
@@ -820,6 +820,88 @@ void jit_select_emitter::emit_isa(const std::vector<size_t> &in_vec_idxs, const
...
+                                             h->fadd(dst.s, src.s, 1);
+                                             h->fdiv(dst.s, dst.s, 1);
+                                             h->fsub(dst.s, 1, dst.s);
+                                             h->fmul(dst.s, dst.s, src.s);
+                                           }
+
+
+// SoftSign
+
+jit_softsign_emitter::jit_softsign_emitter(dnnl::impl::cpu::aarch64::jit_generator *host,
```
Comment (eshoguli) [inline src/plugins/intel_cpu/src/emitters/plugin/aarch64/jit_eltwise_emitters.cpp:863]: SoftSign should be here: https://github.com/openvinotoolkit/openvino/issues/24114

**Review by eshoguli — COMMENTED** [review]
good start, some comments

File: src/plugins/intel_cpu/src/emitters/plugin/aarch64/jit_eltwise_emitters.cpp [hunk 3]
```diff
@@ -896,6 +896,48 @@ void jit_select_emitter::emit_isa(const std::vector<size_t> &in_vec_idxs, const
...
+        OV_CPU_JIT_EMITTER_ASSERT(exec_prc_ == ov::element::f32, "unsupported precision: " + exec_prc_.to_string());
+        using TReg = typename dnnl::impl::cpu::aarch64::cpu_isa_traits<isa>::TReg;
+        const TReg src = TReg(in_vec_idxs[0]);
+        const TReg dst = TReg(out_vec_idxs[0]);
+        const TReg aux = TReg(aux_vec_idxs[0]);
+        h->movi(aux.s, 1);
+        h->fadd(dst.s, src.s, 1);
+        h->fdiv(dst.s, dst.s, 1);
+        h->fsub(dst.s, 1, dst.s);
+        h->fmul(dst.s, dst.s, src.s);
```
Comment (eshoguli) [inline src/plugins/intel_cpu/src/emitters/plugin/aarch64/jit_eltwise_emitters.cpp:936]: `SoftPlus` implementation: https://docs.openvino.ai/2024/documentation/openvino-ir-format/operation-sets/operation-specs/activation/softplus-4.html
You can use reference for `x64` from `oneDNN`: https://github.com/openvinotoolkit/oneDNN/blob/6bf3e2356258149cedf4e31a2c41b9105b4a0da0/src/cpu/x64/injectors/jit_uni_eltwise_injector.cpp#L667

File: src/plugins/intel_cpu/src/emitters/plugin/aarch64/jit_eltwise_emitters.cpp [hunk 4]
```diff
@@ -896,6 +896,55 @@ void jit_select_emitter::emit_isa(const std::vector<size_t> &in_vec_idxs, const
...
+        h->vcvt(dst.s, dst.d);
+        h->vfma(dst.s, v0.s, vptrue, v1.s, dst.s);
+        h->vmaxnm(dst.s, dst.s, vmmin.s);
+        h->vcvt(dst.d, dst.s);
+        h->vexp2(dst.d, dst.d);
+        h->vadd(dst.d, dst.d, vone.d);
+        h->vrcps(dst.d, dst.d);
+        h->vrsqrts(dst.d, dst.d);
+        h->vadd(dst.d, dst.d, vzero.d);
+        h->vst1(dst.d, ptr[dst_reg]);
```
Comment (eshoguli) [inline src/plugins/intel_cpu/src/emitters/plugin/aarch64/jit_eltwise_emitters.cpp:943]: 1. You don't need to load anything in `emitter`. By the main concept: everything has been loaded in vector registers already, use registers.
2. Inside emitter you can use only `in_vec_idxs`, `out_vec_idxs` and `aux_vec_idxs` vector registers (their indexes).
3. In `float` implementation use `word` only.
4. Use instrucations from SIMD&FP Instructions from ARM64 instruction set. We don't support ARM32 in OpenVINO snippets.

Note, please, you can use PRs as example:
* [Abs](https://docs.openvino.ai/2023.3/openvino_docs_ops_arithmetic_Abs_1.html) operation: https://github.com/openvinotoolkit/openvino/pull/23692
* [Sigmoid](https://docs.openvino.ai/2023.3/openvino_docs_ops_activation_Sigmoid_1.html) operation, which reuses [Exp emitter](https://github.com/openvinotoolkit/openvino/blob/4ef40f0f0e70abf04f13310f3e7a8f6784667dbf/src/plugins/intel_cpu/src/emitters/plugin/aarch64/jit_eltwise_emitters.cpp#L820): https://github.com/openvinotoolkit/openvino/pull/23065

**Review by eshoguli — COMMENTED** [review]
Comment about implementation

File: src/plugins/intel_cpu/src/emitters/plugin/aarch64/jit_eltwise_emitters.cpp (hunk 4 above)
Comment (Aryan8912) [reply src/plugins/intel_cpu/src/emitters/plugin/aarch64/jit_eltwise_emitters.cpp:943]: Hey @eshoguli please tell me more details about this, I am busy with other things

**Comment by mlukasze** [comment]
hey @Aryan8912, will you have a time to finish this?

**Pull request closed without merging**